lock-passwordless/src/main/java/com/auth0/lock/passwordless/LockPasswordlessActivity.java

```diff
@@ -0,0 +1,385 @@
+import com.auth0.lock.passwordless.event.PasscodeRequestedEvent;
+import com.auth0.lock.passwordless.event.PasscodeSentEvent;
+import com.auth0.lock.passwordless.event.SelectCountryCodeEvent;
+import com.auth0.lock.passwordless.fragment.InProgressFragment;
+import com.auth0.lock.passwordless.fragment.InvalidLinkFragment;
+import com.auth0.lock.passwordless.fragment.MagicLinkLoginFragment;
+import com.auth0.lock.passwordless.fragment.PasscodeLoginFragment;
+import com.auth0.lock.passwordless.fragment.RequestCodeEmailFragment;
+import com.auth0.lock.passwordless.fragment.RequestCodeSmsFragment;
+import com.auth0.lock.util.ActivityUIHelper;
+import com.squareup.otto.Bus;
+import com.squareup.otto.Subscribe;
+
+public class LockPasswordlessActivity extends FragmentActivity {
+
+    private static final String TAG = LockPasswordlessActivity.class.getName();
+
+    public static final int TYPE_EMAIL = 0;
+    public static final int TYPE_SMS   = 1;
+
+    private static final int REQUEST_CODE = 1234;
+
+    private static final String PASSWORDLESS_TYPE_PARAMETER = "PASSWORDLESS_TYPE_PARAMETER";
+    private static final String USERNAME_PARAMETER = "USERNAME_PARAMETER";
+    private static final String USE_MAGIC_LINK_PARAMETER = "USE_MAGIC_LINK_PARAMETER";
+    private static final String IS_IN_PROGRESS_PARAMETER = "IS_IN_PROGRESS_PARAMETER";
+
+    protected AuthenticationAPIClient client;
+    protected Bus bus;
+    Lock lock;
+    private int passwordlessType;
+    private boolean isInProgress;
+    private boolean useMagicLink;
+    private String username;
+    private LoginAuthenticationErrorBuilder errorBuilder;
+
+    public static void showFrom(Activity activity, int passwordlessType, boolean useMagicLink) {
+        Intent intent = new Intent(activity, LockPasswordlessActivity.class);
+        if (passwordlessType != TYPE_EMAIL && passwordlessType != TYPE_SMS) {
+            Log.e(TAG, "Invalid passwordless type, it must be either TYPE_EMAIL or TYPE_SMS");
+            return;
+        }
+        intent.putExtra(PASSWORDLESS_TYPE_PARAMETER, passwordlessType);
+        intent.putExtra(USE_MAGIC_LINK_PARAMETER, useMagicLink);
+        activity.startActivity(intent);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.com_auth0_passwordless_activity_lock_passwordless);
+
+        lock = getLock();
+        client = lock.getAuthenticationAPIClient();
+        bus = lock.getBus();
+
+        if (savedInstanceState == null) {
+            isInProgress = false;
+            Intent intent = getIntent();
+            passwordlessType = intent.getIntExtra(PASSWORDLESS_TYPE_PARAMETER, TYPE_EMAIL);
+            boolean invalidMagicLink = Intent.ACTION_VIEW.equals(intent.getAction());
+            if (invalidMagicLink) {
+                String dataString = intent.getDataString();
+                if (dataString.contains("/sms?code")) {
```

Comment:
It's probably a good idea to move `the link is a valid magic link` to the `LinkParser` class